dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -83,4 +80,11 @@ object TypedColumn {
       selector: Selector.Aux[H, K, V]
     ): Exists[T, K, V] = new Exists[T, K, V] {}
   }
+
+  implicit class OrderedTypedColumnSyntax[T, U: Ordered](col: TypedColumn[T, U]) {
```

Comment:
Here I'm also wondering whether the `Ordered` contraint should be on the `implicit class` or on the `def`s. We might add something like `@implicitNotFound("The type ${U} is not considered ordered by Spark.")` on the type class.